ci/travis/ci.sh

```diff
@@ -2,7 +2,7 @@
 
 { SHELLOPTS_STACK="${SHELLOPTS_STACK-}|$(set +o); set -$-"; } 2> /dev/null  # Push caller's shell options (quietly)
 
-set -exo pipefail
+set -eo pipefail && if [ -n "${OSTYPE##darwin*}" ]; then set -ux; fi  # some options interfere with Travis's RVM on Mac
```

Comment:
just like `builtin cd`, this is not really maintenable. Is there a way to set these in one place? something like a new file
`prelude.sh` and then `source prelude.sh` in each script.

Reply:
Let's chat more about this, I agree it's not great but all the options have downsides. One trouble is that depends on the current directory, which could be different from the script directory. If I did `source "${ROOT_DIR}"/ci/prelude.sh` then that itself would require doing the `cd` business first to get `ROOT_DIR` which defeats the purpose.

It might be worth noting I _do_ hope to move the other scripts into `ci.sh` later, so this line would ideally only be present in this file. In fact I had already done that in this PR at first, but it made it difficult to review (and revert later potentially), so that's why I didn't do that here. But I think all the CI stuff should be in `ci.sh` for multiple reasons (factoring out this bit being being one of them, but not the only one).